js/interface.js

```diff
@@ -2821,4 +2822,41 @@ function initialLoad(initial, timeout) {
 }
 
 // Start
-initLoad = initialLoad(true, 0);
\ No newline at end of file
+initLoad = initialLoad(true, 0);
+
+// @TODO Move function
+function prompt2FACode() {
+  // @TODO Update Fliplet.Modal in fliplet-api for simpler call
+  return Fliplet.Modal.prompt({
+    title: 'Please enter the verification code to verify your login'
+  }).then(function (code) {
+    if (code === null) {
+      return null;
+    }
+
+    if (code === '') {
+      return Fliplet.Modal.alert({
+        message: 'You must enter the verifiaction code to log in'
+      }).then(function () {
+        return prompt2FACode();
+      });
+    }
+
+    return code;
+  });
+}
+
+// Listen for 2FA code when requested
+socket.on('aab.apple.login.2fa', function (data) {
+  // Ask user for code
+  prompt2FACode().then(function (code) {
+    if (code === null) {
+      // @TODO Cancels login
+      return;
+    }
+
+    // Notify requester with the code
+    // @Question What happens when 2FA fails?
+    socket.to(data.clientId).emit('aab.apple.login.2fa.code', code);
```

Comment:
@squallstar How can I detect and handle 2FA login failure?

Reply:
...or success, for that matter.

Reply:
@tonytlwu I'll see if I can add two events for you to listen, like `aab.apple.login.2fa.success` and `aab.apple.login.2fa.failure`